components/TextFields/MDCTextInput.h

```diff
@@ -0,0 +1,77 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights
+ Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+/**
+ The Material Design guideslines have many suggestions for handling text input. The inputs that
+ conform to this protocol have all the API necessary to customize them to those suggestions.
+
+ They are, however, dumb; they do not handle error state nor validation.
+
+ - For handling error states and other Material behaviors use an MDCTextInputBehavior on your inputs
+ - For validation, there are many 3rd party libraries you can use like:
+   - https://github.com/3lvis/Validation
+   - https://github.com/adamwaite/Validator
+ */
+
+@protocol MDCTextInput;
+
+/** Common API for Material Design themed text inputs. */
+@protocol MDCTextInput <NSObject>
+
+/** A Boolean value indicating whether the text field is currently in edit mode. */
+@property(nonatomic, readonly, getter=isEditing) BOOL editing;
+
+/** The text displayed in the text input. */
+@property(nonatomic, nullable, copy) NSString *text;
+
+/** The color of the text in the input. */
+@property(nonatomic, nullable, strong) UIColor *textColor;
```

Comment:
I was wondering that myself. I originally had copy there but wavered and went with the more conservative option. As my example I checked the header of UITextField. textColor is not copy.

But there are times when the Apple API is not perfect. Should we be? Or would that deviation be bad for our users?